### PR TITLE
refactor: cleanup debug component

### PR DESCRIPTION
**Pull request by MatthijsSmets**
Major cleanup of code, css classes and element ids.

### CODE REVIEW

File: src/app/debug/debug-tab.service.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  providedIn: 'root',
+})
+export class DebugTabService {
+  private refreshSubject: Subject<void> = new Subject();
+  refresh$: Observable<void> = this.refreshSubject.asObservable();
+
+  refresh(): void {
+    this.refreshSubject.next();
+  }
+}
```
Comment (ricklue) [inline src/app/debug/debug-tab.service.ts:14]: Is this pr also part of the refresh issue? Because I see here a new refresh subject

Comment (MatthijsSmets) [reply src/app/debug/debug-tab.service.ts:14]: I created it but haven't implemented it yet, implementation will be done in my next project

File: src/app/debug/table/table.component.css [hunk 2]
```diff
@@ -79,19 +78,14 @@ th {
 
 .progress-report-timout {
```
Comment (MarkvdVorst) [inline src/app/debug/table/table.component.css:79]:
```suggestion
.progress-report-timeout {
```
Small typo. See below for reference in table template

File: src/app/debug/table/table.component.html [hunk 3]
```diff
@@ -111,18 +96,29 @@ <h4 class="progress-report-timout">[One or more reports are in progress for more
           </button>
         }
       }
-      <app-active-filters class="ml-2" [activeFilters]="currentFilters" (click)="toggleFilter()"></app-active-filters>
+      <app-active-filters class="ml-2" [activeFilters]="tableSettings.currentFilters" (click)="toggleFilter()"></app-active-filters>
     </div>
-    <div data-cy-debug="table" class="table-responsive" id="metadataTable">
-      <table data-cy-debug="tableBody" mat-table class="table mb-0" matSort [style.font-size]="getFontSize()" [dataSource]="tableDataSource">
+    @if (hasTimedOut) {
+      <h4 class="progress-report-timout">[One or more reports are in progress for more
```
Comment (MarkvdVorst) [inline src/app/debug/table/table.component.html:102]:
```suggestion
      <h4 class="progress-report-timeout">[One or more reports are in progress for more
```
typo fix for previous comment

File: src/app/debug/table/table.component.ts [hunk 4]
```diff
@@ -272,30 +294,29 @@ export class TableComponent implements OnInit, OnDestroy {
...
+  toggleSelectAll(): void {
+    if (this.amountOfSelectedReports === this.tableSettings.reportMetadata.length) {
+      this.setCheckedForAllReports(false);
+      this.amountOfSelectedReports = 0;
+    } else {
+      this.setCheckedForAllReports(true);
+      this.amountOfSelectedReports = this.tableSettings.reportMetadata.length;
     }
-    return true;
   }
```
Comment (MarkvdVorst) [inline src/app/debug/table/table.component.ts:315]: It might be a good idea to change the checkboxes to the angular way using mat-checkbox. That way we can just grab the length of the SelectionModel. These methods can be simplified this way.

[example of mat-checkbox](https://stackblitz.com/edit/angular-mat-table-checkbox-select-all-ivt1dw)

**Review by MarkvdVorst — CHANGES_REQUESTED** [review]
A couple suggestions.

File: src/app/debug/table/table.component.ts (hunk 4 above)
Comment (MatthijsSmets) [reply src/app/debug/table/table.component.ts:315]: I wanted to use the mat-checkbox, but I saw it had it's own styling with preset themes. So if you can tell me how to custom style it, I will gladly implement it.

File: src/app/debug/debug-tab.service.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (MatthijsSmets) [reply src/app/debug/debug-tab.service.ts:14]: See my comment above

File: src/app/debug/table/table.component.ts (hunk 4 above)
Comment (MarkvdVorst) [reply src/app/debug/table/table.component.ts:315]: [stackoverflow explanation](https://stackoverflow.com/questions/51984326/how-to-style-a-mat-checkbox-properly)

TL;DR
"To style the non checked box:"
```css
    .mat-checkbox-background{
    background-color: transparent;
    svg path{
        fill: transparent;
    }
    .mat-checkbox-checkmark-path {
         stroke:transparent;
         //stroke is the checked mark only
        }
    }
```

"To style the checked box:"
```css
.mat-checkbox-indeterminate.mat-accent .mat-checkbox-background, .mat-checkbox-checked.mat-accent .mat-checkbox-background {
    background-color: #2cc3f8 !important;
    svg path {
        fill: #2cc3f8 !important;
    }
}
```

File: src/app/debug/table/table.component.css (hunk 2 above)
Comment (MatthijsSmets) [reply src/app/debug/table/table.component.css]: fixed

File: src/app/debug/table/table.component.html (hunk 3 above)
Comment (MatthijsSmets) [reply src/app/debug/table/table.component.html]: fixed

File: src/app/debug/table/table.component.ts (hunk 4 above)
Comment (ricklue) [reply src/app/debug/table/table.component.ts:315]: I think these two methods are not complex enough to use a new component of Angular Material that our project depends on

**Review by ricklue — APPROVED** [review]
